docs/pages/setup/reference/cli.mdx

```diff
@@ -1219,10 +1265,10 @@ $ tctl get [<flags>] [<resource-type/resource-name>],...
 | `--format` | | `yaml, json` or `text` | Output format |
 | `--with-secrets` | none | none | Include secrets in resources like certificate authorities or OIDC connectors |
 
-#### [Global Flags](#tctl-global-flags)
+#### Global flags
 
 These flags are available for all commands `--debug, --config` . Run
-`tctl help <subcommand>` or see the [Global Flags Section](#tctl-global-flags)
+`tctl help <subcommand>` or see the [Global Flags section](#tctl-global-flags).
```

Comment:
Re: the "Dump all resources for backup into state.yaml" in the example below this line - in my testing, this only works for customers using the standalone Teleport, and not for Teleport Cloud.  If you try this with a Teleport Cloud instance as a customer, you get the error "this request can be only executed by an admin", even if you are an admin.  

So you might want to update this example comment to say something like: 
# Dump all resources for backup into state.yaml (customers can run this command on on non-Teleport Cloud instances)

Reply:
@pierrebeaucamp Would you mind adding some context for why this is and if there's a way for Cloud users to run `tctl get all`? 

Is it that `tctl` must be run from a Teleport instance host, rather than a local machine?

From browsing the source, it looks like the error above is returned by [`*ServerWithRoles.GetUsers`](https://github.com/gravitational/teleport/blob/4d99f1c9cfa3d2824e8c7acc5bd4c2426de703d0/lib/auth/auth_with_roles.go#L1684) after checking if the server session/context has the [built-in Admin role](https://github.com/gravitational/teleport/blob/4d99f1c9cfa3d2824e8c7acc5bd4c2426de703d0/lib/auth/auth_with_roles.go#L1688). This seems to be set in [`contextForBuiltinRole`](https://github.com/gravitational/teleport/blob/4d99f1c9cfa3d2824e8c7acc5bd4c2426de703d0/lib/auth/permissions.go#L682-L683) while authorizing a user, whose name is provided as the [value](https://github.com/gravitational/teleport/blob/4d99f1c9cfa3d2824e8c7acc5bd4c2426de703d0/lib/auth/permissions.go#L154-L155) of the `teleport-user` key in the `context.Context` passed to `Authorize`. The `teleport-user` key is assigned in two `*Middleware` methods based on the return value of [`*Middleware.GetUser`](https://github.com/gravitational/teleport/blob/4d99f1c9cfa3d2824e8c7acc5bd4c2426de703d0/lib/auth/middleware.go#L425-L426), which uses the client cert to discern the user's name and role. It looks like the Admin role is [added](https://github.com/gravitational/teleport/blob/4d99f1c9cfa3d2824e8c7acc5bd4c2426de703d0/integration/helpers.go#L213-L222) to a host cert when a Teleport process is started—is that why a Cloud user wouldn't be identified as having the Admin role?

Thanks!

Reply:
I only see the error printed in [`*ServerWithRoles.GetUsers`](https://github.com/gravitational/teleport/blob/506da87a96c5bca1249cca99b852eeb4301877cc/lib/auth/auth_with_roles.go#L1678) and [`*ServerWithRoles.GetUser`](https://github.com/gravitational/teleport/blob/506da87a96c5bca1249cca99b852eeb4301877cc/lib/auth/auth_with_roles.go#L1709)—does `tctl get users` also return that error for you? Can you print resources besides users?

Reply:
Thanks again for testing this! I've suggested adding our new `ScopedBlock` component to show `tctl get all` in the code snippet only for self-hosted users.